src/Tools/Microsoft.VisualStudio.Templates.Editorconfig.Wizard/Generator/RoslynEditorConfigFileGenerator.cs

```diff
@@ -61,185 +27,18 @@ public RoslynEditorConfigFileGenerator(DTE2 dte)
-
-            var getServiceMethod = getServiceGenericMethod.MakeGenericMethod(ioptionServiceType);
-            var optionService = getServiceMethod.Invoke(workspace.Services, null);
-
-            var getOptionsMethod = ioptionServiceType.GetMethod("GetOptions");
-            if (getOptionsMethod is null)
-            {
-                return null;
-            }
-
-            return (OptionSet?)getOptionsMethod.Invoke(optionService, null);
-        }
-        catch (Exception ex)
-        {
-            LogException(ex, "Unable to get OptionSet from roslyn");
-            return null;
-        }
-
-    }
-
-    private (bool success, ImmutableArray<(string feature, ImmutableArray<IOption> options)>) GetEditorConfigOptions(string language)
-    {
-        try
-        {
-            var codeStylePageType = GetCodePage(language);
-            Type[] targetParams = new Type[] { };
-            var getEditorConfigOptionsMethod = codeStylePageType?.GetMethod("GetEditorConfigOptions", NonPublicBindingFlags, binder: null, types: targetParams, modifiers: null);
-            if (getEditorConfigOptionsMethod is null)
-            {
-                return (false, default);
-            }
-
-            return (true, (ImmutableArray<(string feature, ImmutableArray<IOption> options)>)getEditorConfigOptionsMethod.Invoke(null, null)!);
-        }
-        catch (Exception ex)
-        {
-            LogException(ex, "Unable to get visual studio options from tools->options page");
-            return (false, default(ImmutableArray<(string feature, ImmutableArray<IOption> options)>));
-        }
-
-        Type? GetCodePage(string language)
-        {
-            var languageName = language == LanguageNames.CSharp ? "CSharp" : "VisualBasic";
-            var codePageTypeName = $"Microsoft.VisualStudio.LanguageServices.{languageName}.Options.Formatting.CodeStylePage";
-            var langueServiceAssembly = GetLanguageServiceAssembly(language);
-            return langueServiceAssembly.GetType(codePageTypeName);
-        }
 
-        Assembly GetLanguageServiceAssembly(string language)
-        {
-            return language switch
-            {
-                LanguageNames.CSharp => CSharpLanguageServiceAssembly,
-                LanguageNames.VisualBasic => VisualBasicLanguageServiceAssembly,
-                _ => throw new ArgumentException(nameof(language))
-            };
-        }
-    }
-
-    private string? Generate(ImmutableArray<(string feature, ImmutableArray<IOption> options)> groupedOptions, OptionSet optionSet, string language)
+    public string? Generate(string language)
     {
         try
         {
-            LogEvents(EventId.Options, new OptionsInfo(groupedOptions, optionSet));
-            return (string?)_generatorMethod.Value?.Invoke(null, new object[] { groupedOptions, optionSet, language });
+            var editorConfigGenerator = _serviceProvider.GetService(typeof(IEditorConfigGenerator)) as IEditorConfigGenerator;
```

Comment:
```suggestion
            var editorConfigGenerator = (IEditorConfigGenerator)_serviceProvider.GetService(typeof(IEditorConfigGenerator));
```

The "as" cast means this becomes undebuggable if it ever fails.

Reply:
Thank you!